### ISSUE TITLE
[Warning] UserWarning: Using categorical_feature in Dataset.

### ISSUE THREAD

**Issue by ucasiggcas** [microsoft/LightGBM#3718]
hi,dear all
I tried according to the [link docs](https://lightgbm.readthedocs.io/en/latest/Python-Intro.html), but has the warning, could you pls help me ?
**train_data = lgb.Dataset(data, label=label, feature_name=['c1', 'c2', 'c3'], categorical_feature=['c3'])**
```
Warning (from warnings module):
  File "D:\python\lib\site-packages\lightgbm\basic.py", line 1291
    warnings.warn('Using categorical_feature in Dataset.')
UserWarning: Using categorical_feature in Dataset.
```

**Comment by StrikerRUS** [comment]
Hi @ucasiggcas !
Everything is fine! This warning is to make you confident that categorical features are not lost. Refer to 
 https://github.com/microsoft/LightGBM/issues/3625#issuecomment-738758114.

**Comment by domkm** [comment]
This is extremely odd. I've never come across a library that logs a warning when things are working correctly. I'd recommend at least updating the message to make it clear to the user that nothing is wrong.

**Comment by jameslamb** [comment]
This is an open source project, you're welcome to propose a pull request changing its behavior.

**Comment by domkm** [comment]
Apologies if my tone was impolite. It was unintentional but, upon rereading, I believe it may have been.

Would a PR to remove this warning be merged?

**Comment by jameslamb** [comment]
Thanks for saying that!

> Would a PR to remove this warning be merged?

Definitely, I agree with your assessment. But I looked tonight and found that actually...this warning has already been removed from `lightgbm`.

This issue is a duplicate of #3379, and the warning was removed in #4768 (November 2021).

Unfortunately, most changes merged to `master` after October 27, 2021 still haven't made it into a release. So if you `pip install lightgbm`, you may still experience this warning. The fix will be in v4.0.0, the next release of LightGBM. You can follow #5153 to be automatically notified when that release comes out. We apologize for the long delay, and are trying to get that release ready as fast as possiblle.

**Comment by domkm** [comment]
That's great! Thank you very much.
